includes/amp-helper-functions.php

```diff
@@ -741,9 +741,16 @@ function amp_add_amphtml_link() {
 	}
 
 	$amp_url = amp_add_paired_endpoint( amp_get_current_url() );
+
 	if ( $amp_url ) {
-		$amp_url = remove_query_arg( QueryVar::NOAMP, $amp_url );
-		printf( '<link rel="amphtml" href="%s">', esc_url( $amp_url ) );
+		$amp_url          = remove_query_arg( QueryVar::NOAMP, $amp_url );
+		$sandboxing_level = amp_get_sandboxing_level();
+
+		if ( 1 === $sandboxing_level || 2 === $sandboxing_level ) {
+			amp_add_alternate_link( $amp_url );
```

Comment:
I just realized that when Mobile Redirection is enabled, this function is now getting called twice, resulting in the same `link` being output at the same time. This being the case, I wonder if instead of adding this here we should instead change the logic to be as follows:

```php
if ( 0 === $sandboxing_level || 3 === $sandboxing_level ) {
	printf( '<link rel="amphtml" href="%s">', esc_url( $amp_url ) );
}
```

Then we can change the `MobileRedirection` service to always print the mobile alternate `link` if mobile redirection is enabled _or_ the sandboxing level is 1 or 2. This will prevent duplicate links from being output and it will bring the number of times this function is called down to 1. In fact, we won't need this helper function anymore.

Reply:
Removed the alternate link helper function and make changes as suggested.